src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3367,14 +3367,46 @@ public override Guid GUID
         {
             get
             {
+                TypeHandle th = GetNativeTypeHandle();
+                if (th.IsTypeDesc || IsArray)
```

Comment:
```suggestion
                if (th.IsTypeDesc || th.AsMethodTable()->IsArray)
```
If you are worried about GetGuid perf, this would make it a bit faster.

Reply:
The `IsArray` doesn't exist on the managed `MethodTable`. I can add it if you'd like though.

Reply:
I think it would be fine to add it. All bits required for it should be exposed already.

Reply:
Nevermind, we should just have this.